imap_processing/hi/l1a/histogram.py

```diff
@@ -0,0 +1,122 @@
+"""Unpack IMAP-Hi histogram data."""
+
+import dataclasses
+
+import numpy as np
+import xarray as xr
+from bitstring import ConstBitStream
+from space_packet_parser.parser import Packet
+
+from imap_processing.cdf.global_attrs import ConstantCoordinates
+from imap_processing.cdf.utils import calc_start_time
+from imap_processing.hi import hi_cdf_attrs
+
+
+def create_dataset(packets: list[Packet]) -> xr.Dataset:
+    """Create dataset for a number of Hi Histogram packets.
+
+    Parameters
+    ----------
+    packets : list[Packet]
+        packet list
+
+    Returns
+    -------
+    xr.dataset
+        dataset with all metadata field data in xr.DataArray
+    """
+    # preallocate the xr.DataArrays for all CDF attributes based on number of packets
+    coords = dict()
+    coords["epoch"] = xr.DataArray(
+        np.empty(len(packets), dtype="datetime64[ns]"),
+        name="epoch",
+        dims=["epoch"],
+        attrs=ConstantCoordinates.EPOCH,
+    )
+    # Histogram data is binned in 90, 4-degree bins
+    # TODO: Confirm whether to define bins by centers or edges. For now centers
+    #    are assumed.
+    coords["angle"] = xr.DataArray(
+        np.arange(2, 360, 4),
+        name="angle",
+        dims=["angle"],
+        attrs=hi_cdf_attrs.hi_hist_l1a_angle_attrs.output(),
+    )
+
+    data_vars = dict()
+    data_vars["esa_step"] = xr.DataArray(
+        np.empty(len(packets), np.dtype("uint8")),
```

Comment:
minor nit: I have a slight preference for being explicit with adding a keyword argument to the empty method call and using the builtin `np.uint8` for consistency with other places we've been using dtypes in the codebase.
```suggestion
        np.empty(len(packets), dtype=np.uint8),
```